test/ReplicatorPropEncTest.cc

```diff
@@ -679,18 +721,60 @@ TEST_CASE_METHOD(ReplicatorPropertyEncryptionTest, "Decryption error", "[Replica
         replicatedDocs.clear();
         resetDBAndReplicator();
         
-        {
-            ExpectingExceptions x;
-            decryptionError = true;
-            replicate();
+        CBLError expectedError = { };
+        CBLError expectedDocReplError = { };
+        bool willRetryToSyncAgain = false;
+        
+        SECTION("503 Error") {
+            decryptionError = {kCBLWebSocketDomain, 503};
+            expectedDocReplError = decryptionError;
+            expectedError = decryptionError;
+            willRetryToSyncAgain = true; // The doc should be retried to sync again
         }
-
+        
+        SECTION("Crypto Error") {
+            decryptionError = {kCBLDomain, kCBLErrorCrypto};
+            expectedDocReplError = decryptionError;
+            expectedError = { };
+        }
+        
+        SECTION("Other Error") {
+            decryptionError = {kCBLDomain, kCBLErrorUnexpectedError};
+            expectedDocReplError = decryptionError;
+            expectedError = { };
+        }
+        
+        CHECK(replicatedDocs.size() == 0);
+        ExpectingExceptions x;
+        replicate(expectedError);
+        
         CHECK(replicatedDocs.size() == 1);
-        CHECK(replicatedDocs[0].error.code == kCBLErrorCrypto);
-        CHECK(replicatedDocs[0].error.domain == kCBLDomain);
+        CHECK(replicatedDocs[0].error.domain == expectedDocReplError.domain);
+        CHECK(replicatedDocs[0].error.code == expectedDocReplError.code);
         
         CBLError error;
         CHECK(!CBLDatabase_GetDocument(db.ref(), "doc1"_sl, &error));
+        
+        // Now try to replicate again with no error:
```

Comment:
Is your test break up the c4 replicator as it goes offline and retry?

Reply:
and so it's not testing the retry logic?

Reply:
It is the same thing b/w retrying and restarting the replicator again. There is no reference in replication logic.

Reply:
Also I'm using a single shot replicator so the replicator in general will retry twice then stopped with the error (in this case 503 error). Then the replicator is restarted again with no error in the encryption callback.

Reply:
I see, after the offline conversation, the first run of replicate() actually reached the retry count and stopped with error.

Reply:
Noted that : I have found that when using DB-to-DB, there is no retry logic implemented in `c4LocalReplicator.hh`. This is a different topic entirely but this means that the replicator is going to stopped after getting any errors.